src/probnum/filtsmooth/_kalman_filter_smoother.py

```diff
@@ -0,0 +1,192 @@
+"""Convenience functions for filtering and smoothing."""
```

Comment:
All in all I think these functions will improve usability a lot! Thanks for adding them.
Aside from the comments I made below: Should we add `Example`s to the respective docstrings?

Reply:
Could be done, but I was too lazy. Should I do it?

Reply:
Agreement: Not today